src/main/kotlin/io/hsar/wh40k/combatsimulator/logic/TacticalActionStrategy.kt

```diff
@@ -2,34 +2,68 @@ package io.hsar.wh40k.combatsimulator.logic
 
 import io.hsar.wh40k.combatsimulator.model.UnitInstance
 import io.hsar.wh40k.combatsimulator.model.World
+import io.hsar.wh40k.combatsimulator.model.unit.BaseStat
 import io.hsar.wh40k.combatsimulator.random.AverageDice
 
 object TacticalActionStrategy : ActionStrategy {
 
-    override fun decideTurnActions(world: World, thisUnit: UnitInstance, possibleActions: Collection<TurnAction>): List<TurnAction> {
+    override fun decideTurnActions(world: World, thisUnit: UnitInstance, possibleActionOptions: Collection<ActionOption>): List<TurnAction> {
         // Shit implementation - units will only ever aim and fire their max damage attack
-        val aimAction = possibleActions
+        val aimAction = possibleActionOptions
                 .find { action ->
-                    action is TurnAction.HalfAim
+                    action is ActionOption.HalfAim
+                }
+                ?.let { action ->
+                    AimAction(action)
                 }
 
-        val maxDamageAttackAction = possibleActions
+        val maxDamageAttackAction = possibleActionOptions
                 .filter { action ->
                     action is DamageCausingAction
                 }
                 .map { eachDamageCausingAction ->
+                    (eachDamageCausingAction as DamageCausingAction)
```

Comment:
You can save some lines by using `filterIsInstance` if you really need the cast.

Reply:
Implmented